supervisor/backups/manager.py

```diff
@@ -727,10 +727,14 @@ async def _validate_location_password(
             and backup.all_locations[location][ATTR_PROTECTED]
         ):
             backup.set_password(password)
-            if not await backup.validate_password():
-                raise BackupInvalidError(
-                    f"Invalid password for backup {backup.slug}", _LOGGER.error
-                )
+            try:
+                if not await backup.validate_password():
+                    raise BackupInvalidError(
+                        f"Invalid password for backup {backup.slug}", _LOGGER.error
+                    )
+            except BackupFileNotFoundError:
+                self.sys_create_task(self.sys_backups.reload(location))
```

Comment:
```suggestion
                self.sys_create_task(self.reload(location))
```
`sys_backups` is how other objects get to backup manager but that's where we are already. Although we could just do this call in the other `except` before raising `BackupFileNotFoundError` and drop this `try` `except` if we prefer, we don't have to schedule the `reload` from here specifically.

Reply:
We don't have the location in the other try except...

Reply:
oh, good point  👍